### PR TITLE
azure-http-specs, add ARM listing available operations case

### PR DESCRIPTION
**Pull request by v-hongli1**
### Context

- fix https://github.com/Azure/cadl-ranch/issues/688
- not covering legacy cases

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
resolve conflict

File: packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts [hunk 1]
```diff
@@ -29,10 +29,40 @@ const validOrder = {
     lastModifiedByType: "User",
   },
 };
+const validAvailableOperations = {
```
Comment (weidongxu-microsoft) [inline packages/azure-http-specs/specs/azure/resource-manager/operation-templates/mockapi.ts:32]: change name

File: packages/azure-http-specs/specs/azure/resource-manager/operation-templates/available-operations.tsp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+using Azure.ResourceManager;
+using TypeSpec.OpenAPI;
+using Spector;
+
+namespace Azure.ResourceManager.OperationTemplates;
+
+@scenario("ListAvailableOperations")
+@scenarioDoc("""
+    Resource GET operation.
+    Expected path: /providers/Azure.ResourceManager.Models.Resources/operations
```
Comment (weidongxu-microsoft) [inline packages/azure-http-specs/specs/azure/resource-manager/operation-templates/available-operations.tsp:19]: Is this path correct?

Comment (v-hongli1) [reply packages/azure-http-specs/specs/azure/resource-manager/operation-templates/available-operations.tsp]: Fixed by the [commit#627a90e](https://github.com/Azure/typespec-azure/pull/1933/commits/627a90e082e6e2488ebd5459716755936fefbd92)

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by msyyc — APPROVED** [review]

**Pull request merged**